scipy/fftpack/realtransforms.py

```diff
@@ -194,6 +200,59 @@ def idct(x, type=2, n=None, axis=-1, norm=None, overwrite_x=False):
     return _dct(x, _TP[type], n, axis, normalize=norm, overwrite_x=overwrite_x)
 
 
+def _get_dct_fun(type, dtype):
+    try:
+        name = {'float64':'ddct%d', 'float32':'dct%d'}[dtype.name]
+    except KeyError:
+        raise ValueError("dtype %s not supported" % dtype)
+    try:
+        f = getattr(_fftpack, name % type)
+    except AttributeError as e:
+        raise ValueError(str(e) + ". Type %d not understood" % type)
+    return f
```

Comment:
Why using strings/getattr instead of directly using the function references, `{'float32': _fftpack.dct_float32, ...}`